examples/gke-tpu-v4-xpk/gke-tpu-v4.yaml

```diff
@@ -0,0 +1,144 @@
+# Copyright 2025 "Google LLC"
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+blueprint_name: gke-tpu-v4
+
+vars:
+  # The following variables should be over-written in the deployment.yaml file.
+  # Your GCP Project ID
+  project_id:
+
+  # This should be unique across all of your Cluster
+  # Toolkit Deployments.
+  deployment_name: gke-tpu-v4
+
+  # The GCP Region used for this deployment.
+  region:
+
+  # The GCP Zone used for this deployment.
+  zone:
+
+  # The number of nodepools to create
+  num_node_pools:
+
+  # The TPU placement topology for pod slice node pool.
+  tpu_topology:
+
+  # The number of nodes to be created in each nodepool
+  static_node_count:
+
+  # Cidr block containing the IP of the machine calling terraform.
+  # To allow all (IAM restrictions still enforced), use 0.0.0.0/0
+  # To allow only your IP address, use <YOUR-IP-ADDRESS>/32
+  authorized_cidr:
+
+  # The name of the compute engine reservation of A4-High nodes in the form of
+  # <project>/<reservation-name>/reservationBlocks/<reservation-block-name>
+  # extended_reservation:
+
+
+deployment_groups:
+- group: primary
+  modules:
+  - id: gke-tpu-v4-net
+    source: modules/network/vpc
+    settings:
+      network_name: $(vars.deployment_name)-net-0
+      mtu: 8896
+      subnetworks:
+      - subnet_name: $(vars.deployment_name)-sub-0
+        subnet_region: $(vars.region)
+        subnet_ip: 192.168.0.0/18
+      secondary_ranges_list:
+      - subnetwork_name: $(vars.deployment_name)-sub-0
+        ranges:
+        - range_name: pods
+          ip_cidr_range: 10.4.0.0/14
+        - range_name: services
+          ip_cidr_range: 10.0.32.0/20
+      firewall_rules:
+      - name: $(vars.deployment_name)-internal-0
+        ranges: [192.168.0.0/16]
```

Comment:
I notice you created a subnet with /18, and here your firewall rule is broader (/16)... is there a reason for that?

Reply:
Took the piece of code from this one: https://github.com/GoogleCloudPlatform/cluster-toolkit/blob/1d9202642be608449ae7032d4c6f057c84ba394c/examples/gke-a3-ultragpu/gke-a3-ultragpu.yaml#L65

I will try to find out the information.

Reply:
I received this information:

All the VPC individual subnet cidrs are combined in the firewall cidr - this allows for cross VPC access.
```
192.168.0.0/18
192.168.64.0/18
192.168.128.0/18
```
P.S this does not fully add up to [192.168.0.0/16](http://192.168.0.0/16) - there's 25% of IPs still left.

Reply:
what am I missing, I see: 
        sub-0: 192.168.0.0/18
        pods: 10.4.0.0/14
        services: 10.0.32.0/20
        
Where are the other VPCs used?

Reply:
Oh my bad, I mixed up the examples. You are right, in this case, firewalls can remain with /18